src/main/java/hudson/plugins/jira/Updater.java

```diff
@@ -127,22 +127,38 @@ static void submitComments(
             boolean useWikiStyleComments, boolean recordScmChanges, String groupVisibility, String roleVisibility) throws RemoteException {
         // copy to prevent ConcurrentModificationException
         List<JiraIssue> copy = new ArrayList<JiraIssue>(issues);
+        StringBuilder issuesWithExceptions = new StringBuilder();
         for (JiraIssue issue : copy) {
-            try {
                 logger.println(Messages.Updater_Updating(issue.id));
-                session.addComment(
+                
+                try {
+                	session.addComment(
                         issue.id,
                         createComment(build, useWikiStyleComments, jenkinsRootUrl, recordScmChanges, issue),
                         groupVisibility, roleVisibility);
-            } catch (RemotePermissionException e) {
-                // Seems like RemotePermissionException can mean 'no permission' as well as
-                // 'issue doesn't exist'.
-                // To prevent carrying forward invalid issues forever, we have to drop them
-                // even if the cause of the exception was different.
-                logger.println("Looks like " + issue.id + " is no valid JIRA issue or you don't have permission to update the issue.\n" +
-                        "Issue will not be updated.\n" + e);
+                } catch (RemotePermissionException e) {
+                    // Seems like RemotePermissionException can mean 'no permission' as well as
+                    // 'issue doesn't exist'.
+                    // To prevent carrying forward invalid issues forever, we have to drop them
+                    // even if the cause of the exception was different.
+                    logger.println("Looks like " + issue.id + " is no valid JIRA issue or you don't have permission to update the issue.\n" +
+                            "Issue will not be updated.\n" + e);
+                } catch (RemoteException e) {
+                	if( 0 < issuesWithExceptions.length()) {
```

Comment:
and according to codestyle it should look like **if (x > 0)**, please fix it in other places as well